File: 算法/03_serial_number.py
class Solution:
    def sortedSquares(self, nums: list[int]) -> list[int]:
        
        left = 0
        right = len(nums)-1
        k = len(nums)-1
        sorted = [0]*len(nums)
        while(k!=-1):
            if(nums[right]**2>nums[left]**2):
                sorted[k] = nums[right]**2
                k-=1
                right-=1
            else:
                sorted[k] = nums[left]**2
                k-=1
                left+=1
        return sorted
if __name__ == "__main__":
    test = [-4,-1,0,3,10]
    sol = Solution()
    sol.sortedSquares(test)

File: 算法/03_serial_number.cpp
#include<iostream>
#include<vector>
using namespace std;
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int left = 0;
        int right= nums.size()-1;
        int k = nums.size()-1;
        vector<int> sorted(nums.size(),0);
        while(k!=-1)
        {
            if(nums[left]*nums[left]>nums[right]*nums[right])
            {
                sorted[k--] = nums[left]*nums[left];
                left++;
            }else
            {
                sorted[k--] = nums[right]*nums[right];
                right--;
            }
        }
        return sorted;
    }
};

int main()
{
    vector<int> test;
    test = {-4,-1,0,3,10};
    Solution sol;
    sol.sortedSquares(test);
    return 0;
}